src/index.js

```diff
@@ -47,34 +47,48 @@ app.post('/interactive-message', (req, res) => {
   const payload = JSON.parse(req.body.payload);
   const token = payload.token;
   const actions = payload.actions;
-  const callback_id = payload.callback_id;
+  const action_id = actions[0].action_id;
+  const action_array = action_id.split(".");
+  var fieldName = action_array[0];
+  const ticket_id = action_array[1];
   const response_url = payload.response_url;
   const user = payload.user;
 
   if (token === process.env.SLACK_VERIFICATION_TOKEN) {
     // Immediately respond to signal success, further updates will be done using `response_url`
     res.send('');
-
-    Ticket.find(callback_id).then((ticket) => {
+    Ticket.find(ticket_id).then((ticket) => {
       debug('interactive message ticket found');
       const action = actions[0];
-      const fieldName = action.name;
-      // Handle either message menu action or message button (Claim)
-      const fieldValue = action.selected_options ? action.selected_options[0].value : user.id;
+      switch (fieldName) {
+        case 'claim':
+          fieldName = 'agent';
+          var fieldValue = user.id;
+          break;
+        case 'agent':
+          var fieldValue = action.selected_user;
+          break;
+        case 'priority':
+          var fieldValue = action.selected_option.value;
+          break;
+
+        default:
```

Comment:
if execution ever gets to `default` (which it shouldn't), it would proceed to set the `fieldName` of the ticket to `undefined` (that's what `fieldValue` is initialized to). is that good behavior? probably not.

instead, maybe emit a warning (using the `debug()` function) and then return so that `ticket.updateField()` doesn't get called?